File: src/main/java/toker/warbandscripts/panel/controller/rest/ItemController.java
package toker.warbandscripts.panel.controller.rest;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.cache.annotation.Cacheable;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import toker.warbandscripts.panel.entity.Item;
import toker.warbandscripts.panel.repository.ItemRepository;

import java.util.List;

@RestController
public class ItemController {

    private ItemRepository itemRepository;

    public ItemController(ItemRepository itemRepository) {
        this.itemRepository = itemRepository;
    }

    @GetMapping("/api/item")
    @Cacheable("items")
    public List<Item> items() {
        return itemRepository.findAll();
    }
}

File: src/main/java/toker/panel/controller/rest/ItemController.kt
package toker.panel.controller.rest

import org.springframework.cache.annotation.Cacheable
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import toker.panel.entity.Item
import toker.panel.repository.ItemRepository

@RestController
class ItemController(private val itemRepository: ItemRepository) {
    @GetMapping("/api/item")
    @Cacheable("items")
    fun items(): List<Item> {
        return itemRepository.findAll()
    }
}